### DISCUSSION TITLE
Is it possible to use React Native with Reactpy without javascript?

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [idom-team/idom#1177] [Question] [answered]
I discovered the project recently, I found the idea of creating apps without using javascript interesting and a question arose as to whether it would be possible to use React Native without javascript?

**Comment by Archmonger** [comment, marked as answer]
There currently isn't support for React Native. Since we're running Python, it's likely we'd need to fork it into a package like `reactpy-native`.

The main issue is right now ReactPy is reliant on a backend framework, which is server-sided. This clashes with the way React Native functions. I am working on decoupling ReactPy from our backends.

After decoupling, running ReactPy as a native app would require a mixture of PyScript and Electron.js. We've created proof of concepts showing ReactPy can work in this configuration, but we haven't had time to fold it into a clean `reactpy-native` package.
